index.js

```diff
@@ -66,10 +68,20 @@ class Pipe2Jpeg extends Transform {
    * @private
    */
   _transform(chunk, encoding, callback) {
-    const chunkLength = chunk.length;
+    let chunkLength = chunk.length;
     let pos = 0;
     while (true) {
       if (this._size) {
+        const lastChunk = this._chunks[this._chunks.length - 1];
+        const lastByte = lastChunk[lastChunk.length - 1];
+        if (lastByte === _EOI[0] && chunk[0] === _EOI[1]) {
+          // EOI was split across chunks, remove it from the previous chunk and add it to this one
+          this._chunks[this._chunks.length - 1] = lastChunk.slice(0, lastChunk.length - 1);
+          this._size -= 1;
```

Comment:
Should we also decrement `_lastSize` here?

Reply:
I don't think so? `_lastSize` refers to the last size of the jpeg, which includes all chunks. Maybe I should name it `_lastJpegSize`